### PR TITLE
Adding Conv2DLSTM

### PR DESCRIPTION
**Pull request by mbz**
Mostly ported from Haiku implementation.

### CODE REVIEW

**Comment by avital** [comment]
Hi @mbz, thanks for the PR! Can you please add a test, similar to the ones we have in RecurrentTest for LSTMs? https://github.com/google/flax/blob/master/tests/nn_test.py#L623

**Comment by mbz** [comment]
Added a test.
Added to nn.__init__
Renamed to ConvLSTM.

File: flax/nn/recurrent.py [hunk 1]
```diff
@@ -193,3 +195,75 @@ def initialize_carry(rng, batch_dims, size, init_fn=initializers.zeros):
...
+    input_to_hidden = linear.Conv.partial(
+        features=4*features, kernel_size=kernel_size, name="ih")
+
+    hidden_to_hidden = linear.Conv.partial(
+        features=4*features, kernel_size=kernel_size, name="hh")
+
+    gates = input_to_hidden(inputs) + hidden_to_hidden(h)
+    i, g, f, o = jnp.split(gates, indices_or_sections=4, axis=-1)
+
+    f = jnn.sigmoid(f + 1)
```
Comment (jheek) [inline flax/nn/recurrent.py:250]: nit: you can just use `nn.sigmoid` here avoiding the jnn import

File: flax/nn/recurrent.py [hunk 2]
```diff
@@ -193,3 +195,75 @@ def initialize_carry(rng, batch_dims, size, init_fn=initializers.zeros):
...
+    new_c = f * c + jnn.sigmoid(i) * jnp.tanh(g)
+    new_h = jnn.sigmoid(o) * jnp.tanh(new_c)
```
Comment (jheek) [inline flax/nn/recurrent.py:252]: here as well

File: flax/nn/recurrent.py [hunk 3]
```diff
@@ -31,6 +31,8 @@
 from . import initializers
 from . import linear
 
+from jax import nn as jnn
```
Comment (jheek) [inline flax/nn/recurrent.py:34]: this import shouldn't be needed see below

**Review by jheek — CHANGES_REQUESTED** [review]

**Comment by avital** [comment]
@jheek can you please give this a final review and if it's OK merge it? (or if you want to make minor changes you can push straight to the mbz/flax:patch-1 as well)

**Review by jheek — APPROVED** [review]
looks good to me
